es/provider.go

```diff
@@ -250,21 +256,30 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 }
 
 func awsHttpClient(region string, d *schema.ResourceData) *http.Client {
-	creds := awscredentials.NewChainCredentials([]awscredentials.Provider{
-		&awscredentials.StaticProvider{
-			Value: awscredentials.Value{
-				AccessKeyID:     d.Get("aws_access_key").(string),
-				SecretAccessKey: d.Get("aws_secret_key").(string),
-				SessionToken:    d.Get("aws_token").(string),
-			},
+	aws_access_key_id := d.Get("aws_access_key").(string)
+	aws_secret_access_key := d.Get("aws_secret_key").(string)
+	aws_session_token := d.Get("aws_token").(string)
+	aws_profile := d.Get("aws_profile").(string)
+
+	sessOpts := awssession.Options{
+		Config: aws.Config{
+			Region: aws.String(region),
 		},
-		&awscredentials.EnvProvider{},
```

Comment:
Just double checking, the docs say (https://docs.aws.amazon.com/sdk-for-go/api/aws/session/#hdr-Credential_and_config_loading_order):
> The Environment variables for credentials will have precedence over shared config even if SharedConfig is enabled. To override this behavior, and use shared config credentials instead specify the session.Options.Profile

It seems like we'd still need to specify these other providers? I didn't read the underlying code or try it, can you confirm?

Reply:
Sorry - I had planned to write some tests around this to ensure all credential sources are used correctly.  You may be right there - if env vars are specified, then that could override the session profile.

I will try to get this done asap.

Reply:
OK, I understand your concern now.  So, without specifying the providers and just passing in what was created by https://docs.aws.amazon.com/sdk-for-go/api/aws/session/#NewSessionWithOptions (ie: defaults) - default credentials provider chain is used.  The code I added basically only overrides the defaults for the 2 cases: access keys configured on the provider + named profile specified on the provider.  Otherwise it uses the defaults.

I added some tests to make sure this is tested (it only gets as far as confirming that Env credentials/profiles are used if no provider config is specified.  There's no tests for checking EC2 metadata though - didn't feel this would be necessary)